src/java/org/apache/cassandra/db/virtual/SettingsTable.java

```diff
@@ -108,6 +115,28 @@ else if (overrides.containsKey(f.getName()))
             if (value.getClass().isArray())
                 value = Arrays.toString((Object[]) value);
             result.row(f.getName()).column(VALUE, value.toString());
+
+            if(ANNOTATED_FIELDS.containsKey(f.getName()))
+            {
+                if (CassandraDuration.class.equals(value.getClass()))
+                {
+                    CassandraDuration quantity = (CassandraDuration) value;
+                    result.row(f.getAnnotation(Replaces.class).oldName()).column(VALUE,quantity.quantityToString());
+                }
+
+                if (DataStorage.class.equals(value.getClass()))
+                {
+                    DataStorage quantity = (DataStorage) value;
+                    result.row(f.getAnnotation(Replaces.class).oldName()).column(VALUE,quantity.quantityToString());
+                }
+
+                if (BitRate.class.equals(value.getClass()))
+                {
+                    BitRate quantity = (BitRate) value;
+                    result.row(f.getAnnotation(Replaces.class).oldName()).column(VALUE,quantity.quantityToString());
+                }
```

Comment:
For instance `permissions_validity_in_ms` defaults to `2000` before this patch, and now it's a duration of "2s". Does this mean the row will be `permissions_validity_in_ms, 2` in after the patch where it was `permissions_validity_in_ms, 2000` before? If so, it seems like we'd have to clean those up in `Config`?

Reply:
At first we were relying, yes, but now I am going to change that.
This will be changed to acknowledge the converter for both JMX and VT, you can ignore it for now. 
It is already agreed it should be changed, it was acknowledged by David in the PR and I think I left a note on the ticket.

We had a broader discussion around the future of VTs and about another ticket which is open for JMX and metrics.
Agreed to add the converters as part of this ticket to ensure we don't rely on Config. The follow up tickets might take time so better to do it here.

Reply:
Nice. Thanks for getting me up to speed.